UnityProject/Assets/Scripts/Sprite Handler/SpriteHandler.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using UnityEditor;
+
+
+///	<summary>
+///	Handles sprite syncing between server and clients and contains a custom animator
+///	</summary>
+public class SpriteHandler : NetworkBehaviour
+{
+	public SpriteRenderer spriteRenderer;
+
+	public List<SpriteList> spriteList = new List<SpriteList>();
+	[SyncVar(hook = nameof(SyncSprite))] public int spriteIndex;
+
+	private List<SpriteInfo> infoList = new List<SpriteInfo>();
+	private SpriteJson spriteJson;
+	private int animationIndex = 0;
+	private float timeElapsed = 0;
+	private float waitTime;
+
+	private void Awake() {
+		infoList.Add(new SpriteInfo());
+	}
+
+	public override void OnStartClient()
+	{
+		SyncSprite(spriteIndex);
+	}
+
+	public void UpdateMe()
+	{
+		timeElapsed += Time.deltaTime;
+		if(timeElapsed >= waitTime)
+		{
+			animationIndex++;
+			if(animationIndex >= spriteJson.Frames_Of_Animation)
+			{
+				animationIndex = 0;
+			}
+			SetSprite(infoList[animationIndex]);
+		}
+	}
+
+	void SetSprite(SpriteInfo animationStills)
+	{
+		timeElapsed = 0;
+		waitTime = animationStills.waitTime;
+		spriteRenderer.sprite = animationStills.sprite;
+	}
+
+	void SetSpriteList(List<Sprite> newSprites)
+	{
+		if (newSprites.Count > 1)
+		{
+			LoadJson(AssetDatabase.GetAssetPath(newSprites[0]));
```

Comment:
Since this is part of the UnityEditor Namespace
https://docs.unity3d.com/ScriptReference/UnityEditor.html
https://docs.unity3d.com/ScriptReference/AssetDatabase.html

It doesn't work in builds, So you have to keep a note of that if you want to do it in runtime, You'll only be able to run this in the Editor

Reply:
I would just convert this to Resources Load if its meant to run at edit time. All you can make a CustomEditor for this component that can do all that at edit time and cache the reference.